### PR TITLE
Bug 2053006: Resource id fetch optimised

### PR DESCRIPTION
**Pull request by gayathrimenath-ibm**

### CODE REVIEW

**Review by akgunjal — APPROVED** [review]
LGTM

**Comment by jsafrane** [comment]
/retitle Bug 2053006: Resource id fetch optimised

File: pkg/controller/secret/secretsync.go [hunk 1]
```diff
@@ -193,18 +193,23 @@ func defaultGetResourceID(resourceName, accountID, apiKey string) (string, error
...
 	listResourceGroupsOptions.SetAccountID(accountID)
-	resourceGroupList, _, err := serviceClient.ListResourceGroups(listResourceGroupsOptions)
+	listResourceGroupsOptions.SetName(resourceName)
+	resourceGroupList, detailedResponse, err := serviceClient.ListResourceGroups(listResourceGroupsOptions)
 
 	if err != nil {
 		return "", err
 	}
+
+	transactionId := detailedResponse.Headers.Get("Transaction-Id")
```
Comment (arahamad) [inline pkg/controller/secret/secretsync.go:203]: can you put these lines after 213 line, as its only required if it failed

File: pkg/controller/secret/secretsync.go [hunk 2]
```diff
@@ -193,18 +193,23 @@ func defaultGetResourceID(resourceName, accountID, apiKey string) (string, error
...
 		for _, v := range resourceGroupList.Resources {
-			if *v.Name == resourceName {
+			if v.Name!= nil && *v.Name == resourceName {
 				resourceID := *v.ID
 				return resourceID, nil
 			}
 		}
 	}
-	return "", fmt.Errorf("Resource %s not found for given g2Credentials", resourceName)
+	return "", fmt.Errorf("Resource %s could not locate at the moment, controller will retry after %s.", resourceName, util.Resync.String())
```
Comment (arahamad) [inline pkg/controller/secret/secretsync.go:214]: can we return error message like `fmt.Errorf("Resource group '%s' could not locate at the moment, controller will retry after %s.", resourceName, util.Resync.String())`

**Comment by jsafrane** [comment]
@gayathrimenath-ibm can you please add some explanation "why" to the commit message? Like it should avoid any paging issues in the cloud API.

**Comment by jsafrane** [comment]
From ci/prow/verify

> verify-gofmt failed - please run `make update-gofmt`

**Pull request merged**